### PR TITLE
[MM-36075] Replace ASGs for EKS with managed nodes

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Replace ASGs for EKS with managed nodes
 
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36075

### CODE REVIEW

File: terraform/aws/modules/cluster/worker_asg.tf [hunk 1]
```diff
@@ -83,3 +25,68 @@ data:
...
+  name        = "cnc_cluster_launch_template"
+  description = "CnC cluster nodes launch template"
+
+  vpc_security_group_ids = [aws_security_group.worker-sg.id]
+
+  block_device_mappings {
+    device_name = "/dev/xvda"
+
+    ebs {
+      volume_size = 20
```
Comment (spirosoik) [inline terraform/aws/modules/cluster/worker_asg.tf:40]: let's use a variable for this

File: terraform/aws/modules/cluster/worker_asg.tf [hunk 2]
```diff
@@ -83,3 +25,68 @@ data:
...
+      volume_type = "gp2"
```
Comment (spirosoik) [inline terraform/aws/modules/cluster/worker_asg.tf:41]: same for this

File: terraform/aws/modules/cluster/worker_asg.tf [hunk 3]
```diff
@@ -83,3 +25,68 @@ data:
...
+
+    tags = {
+      Name = "cnc-cluster-nodes"
+    }
+  }
+}
+
+resource "aws_eks_node_group" "general_nodes_eks_cluster_ng" {
+  cluster_name    = aws_eks_cluster.cluster.name
+  node_group_name = "general-nodes"
```
Comment (spirosoik) [inline terraform/aws/modules/cluster/worker_asg.tf:61]: do we want to give a dynamic map from a variable and prepend or append `-nodes` or any other convention we want

File: terraform/aws/modules/eks-cluster/worker_asg.tf [hunk 4]
```diff
@@ -67,3 +25,67 @@ data:
         - system:nodes
 CONFIGMAPAWSAUTH
 }
+
+resource "aws_launch_template" "cluster_nodes_eks_launch_template" {
```
Comment (spirosoik) [inline terraform/aws/modules/eks-cluster/worker_asg.tf:29]: seems like a duplicate right? Why not make this a mini module reusable ?

File: terraform/aws/modules/eks-cluster/worker_iam.tf [hunk 5]
```diff
@@ -27,7 +27,7 @@ resource "aws_iam_policy" "teleport_policy" {
 {
     "Version": "2012-10-17",
     "Statement": [
-        {   
+        {
```
Comment (spirosoik) [inline terraform/aws/modules/eks-cluster/worker_iam.tf:30]: same here as above.

**Review by spirosoik — APPROVED** [review]
no blockers, just minor improvements if we want to make. it will be nice to avoid the duplication and have a reusable moduelf or this

**Review by angeloskyratzakos — APPROVED** [review]
LGTM

File: terraform/aws/modules/eks-cluster/worker_asg.tf [hunk 6]
```diff
@@ -67,3 +25,67 @@ data:
...
+  description = "CnC cluster nodes launch template"
+
+  vpc_security_group_ids = [aws_security_group.worker-sg.id]
+
+  block_device_mappings {
+    device_name = "/dev/xvda"
+
+    ebs {
+      volume_size = 20
+      volume_type = "gp2"
```
Comment (stafot) [inline terraform/aws/modules/eks-cluster/worker_asg.tf:40]: same here as above, let's do them variables

File: terraform/aws/modules/eks-managed-node-groups/variables.tf [hunk 7]
```diff
@@ -0,0 +1,29 @@
+variable "vpc_security_group_ids" {}
```
Comment (spirosoik) [inline terraform/aws/modules/eks-managed-node-groups/variables.tf:1]: no blocker, some description for each field will be nice :)

**Review by spirosoik — APPROVED** [review]
great job

File: terraform/aws/modules/eks-managed-node-groups/variables.tf (hunk 7 above)
Comment (stylianosrigas) [reply terraform/aws/modules/eks-managed-node-groups/variables.tf]: Done :)

**Pull request merged**